app/post.go

```diff
@@ -440,6 +440,82 @@ func (a *App) UpdatePost(post *model.Post, safeUpdate bool) (*model.Post, *model
 	}
 }
 
+func (a *App) RethreadPost(post *model.Post, safeUpdate bool) (*model.Post, *model.AppError) {
+	post.SanitizeProps()
+
+	var oldPost *model.Post
+	var thread *model.PostList
+	if result := <-a.Srv.Store.Post().Get(post.Id); result.Err != nil {
+		return nil, result.Err
+	} else {
+		oldPost = result.Data.(*model.PostList).Posts[post.Id]
+		thread = result.Data.(*model.PostList)
+		if oldPost == nil {
+			err := model.NewAppError("UpdatePost", "api.post.rethread_post.find.app_error", nil, "id="+post.Id, http.StatusBadRequest)
+			return nil, err
+		}
+
+		if oldPost.DeleteAt != 0 {
+			err := model.NewAppError("UpdatePost", "api.post.rethread_post.permissions_details.app_error", map[string]interface{}{"PostId": post.Id}, "", http.StatusBadRequest)
+			return nil, err
+		}
+
+		if oldPost.IsSystemMessage() {
+			err := model.NewAppError("UpdatePost", "api.post.rethread_post.system_message.app_error", nil, "id="+post.Id, http.StatusBadRequest)
+			return nil, err
+		}
+
+		if a.License() != nil {
+			if *a.Config().ServiceSettings.PostEditTimeLimit != -1 && model.GetMillis() > oldPost.CreateAt+int64(*a.Config().ServiceSettings.PostEditTimeLimit*1000) && post.Message != oldPost.Message {
+				err := model.NewAppError("UpdatePost", "api.post.rethread_post.permissions_time_limit.app_error", map[string]interface{}{"timeLimit": *a.Config().ServiceSettings.PostEditTimeLimit}, "", http.StatusBadRequest)
+				return nil, err
+			}
+		}
+	}
+
+	newPost := &model.Post{}
+	*newPost = *oldPost
+	
+	if newPost.RootId == "" && post.RootId != "" && post.RootId != newPost.Id{
+    	if _, ok := thread.Posts[post.Id]; ok && len(thread.Posts) == 1 {
+    		newPost.RootId = post.RootId
+    	}
```

Comment:
looks like something is wrong with the indentation or this one is extraneous (also, this line is using space indent)